src/MockAWS.jl

```diff
@@ -24,41 +21,67 @@ macro patches(service::Symbol)
     return Expr(:toplevel, :(Main.$(service).include($service_name)))
 end
 
-function _get_aws_sdk_js_files(repo_name::String, auth::GitHub.OAuth2)
-    function _filter_latest_service_version(services::AbstractArray)
-        seen_services = Set{String}()
-        latest_versions = OrderedDict[]
 
-        for service in reverse(services)
-            service_name, _ = _get_service_and_version(service["path"])
+"""
+Get all aws/aws-sdk-js files
+"""
+function _get_aws_sdk_js_files(auth::GitHub.OAuth2)
+    master_tree = tree(REPO_NAME, "master"; auth=auth)
+    apis_sha = [t for t in master_tree.tree if t["path"]=="apis"][1]["sha"]
+    files = tree(REPO_NAME, apis_sha)
+
+    return files.tree
+end
 
-            if !(service_name in seen_services)
-                push!(seen_services, service_name)
-                push!(latest_versions, service)
-            end
-        end
 
-        return latest_versions
+"""
+Filter only examples.json files
+"""
+function _get_example_files(files::Vector, auth::GitHub.OAuth2)
+    example_files = filter(f -> endswith(f["path"], ".examples.json"), files)
+
+    seen_services = Set{String}()
+    latest_versions = OrderedDict[]
+
+    for service in reverse(example_files)
+        service_name, _ = @mock _get_service_and_version(service["path"], files, auth)
+
+        if !(service_name in seen_services)
+            push!(seen_services, service_name)
+            push!(latest_versions, service)
+        end
     end
 
-    master_tree = tree(repo_name, "master"; auth=auth)
-    apis_sha = [t for t in master_tree.tree if t["path"]=="apis"][1]["sha"]
-    files = tree(repo_name, apis_sha)
-    files = files.tree
+    return latest_versions
+end
 
-    filter!(f -> endswith(f["path"], ".examples.json"), files)
 
-    return _filter_latest_service_version(files)
+"""
+Find the SHA hash for a give filename
+"""
+function _get_sha(filename::String, files::Vector)
+    for file in files
+        if file["path"] == filename
+            return file["sha"]
+        end
+    end
 end
 
-function _get_service_and_version(filename::String)
+
+"""
+Get service name and version from normal.json file
+"""
+function _get_service_and_version(filename::String, files::Vector, auth::GitHub.OAuth2)
     try
-        # Remove ".example.json" suffix
-        service_and_version = join(split(filename, '.')[1:end-2],'.')
+        filename = replace(filename, "examples"=>"normal")
```

Comment:
Why do you replace examples with normal?

Reply:
I mentioned this in the PR description. But we use the `serviceId` in `AWS.jl` to create the high-level wrapper file name, for example its `Secrets Manager` which then becomes `secrets_manager.jl`. 

The `examples.json` filename doesn't have this information, nor do the contents of it. So instead just looking it up by using the `metadata` JSON block from the `normal.json` file.

It's a bit inefficient as a better way might be to just convert `AWS.jl` to use something else, but that's a major version update which I don't want to do (at least yet).

Reply:
I see, thanks! I think this is fine for now.